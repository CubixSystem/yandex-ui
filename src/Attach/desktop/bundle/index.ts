export * from '../../Attach.bundle/desktop';
