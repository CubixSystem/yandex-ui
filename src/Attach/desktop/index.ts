export * from '../Attach@desktop';
export * from '../_size/Attach_size_l';
export * from '../_size/Attach_size_m';
export * from '../_size/Attach_size_s';
