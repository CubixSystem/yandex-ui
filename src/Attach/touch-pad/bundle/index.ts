export * from '../../Attach.bundle/touch-pad';
