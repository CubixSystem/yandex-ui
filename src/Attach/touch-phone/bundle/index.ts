export * from '../../Attach.bundle/touch-phone';
