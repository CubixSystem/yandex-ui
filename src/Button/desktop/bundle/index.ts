import { compose, composeU, ExtractProps } from '@bem-react/core';

import { Button as ButtonDesktop } from '../../Button@desktop';
// _baseline
import { withBaseline } from '../../_baseline/Button_baseline';
// _pin
import { withPinBrickBrick } from '../../_pin/Button_pin_brick-brick';
import { withPinBrickCircle } from '../../_pin/Button_pin_brick-circle';
import { withPinBrickClear } from '../../_pin/Button_pin_brick-clear';
import { withPinBrickRound } from '../../_pin/Button_pin_brick-round';
import { withPinCircleBrick } from '../../_pin/Button_pin_circle-brick';
import { withPinCircleCircle } from '../../_pin/Button_pin_circle-circle';
import { withPinCircleClear } from '../../_pin/Button_pin_circle-clear';
import { withPinClearBrick } from '../../_pin/Button_pin_clear-brick';
import { withPinClearCircle } from '../../_pin/Button_pin_clear-circle';
import { withPinClearClear } from '../../_pin/Button_pin_clear-clear';
import { withPinClearRound } from '../../_pin/Button_pin_clear-round';
import { withPinRoundBrick } from '../../_pin/Button_pin_round-brick';
import { withPinRoundClear } from '../../_pin/Button_pin_round-clear';
// _size
import { withSizeL } from '../../_size/Button_size_l';
import { withSizeM } from '../../_size/Button_size_m';
import { withSizeS } from '../../_size/Button_size_s';
// _theme
import { withThemeAction } from '../../_theme/Button_theme_action@desktop';
import { withThemeClear } from '../../_theme/Button_theme_clear@desktop';
import { withThemeLink } from '../../_theme/Button_theme_link';
import { withThemeNormal } from '../../_theme/Button_theme_normal@desktop';
import { withThemePseudo } from '../../_theme/Button_theme_pseudo@desktop';
import { withThemeRaised } from '../../_theme/Button_theme_raised';
import { withThemeWebSearch } from '../../_theme/Button_theme_websearch';
// _type
import { withTypeLink } from '../../_type/Button_type_link';
import { withTypeSubmit } from '../../_type/Button_type_submit';
// _view
import { withViewDefault } from '../../_view/Button_view_default@desktop';
import { withViewAction } from '../../_view/Button_view_action@desktop';
import { withViewClear } from '../../_view/Button_view_clear@desktop';
import { withViewLink } from '../../_view/Button_view_link@desktop';
import { withViewPseudo } from '../../_view/Button_view_pseudo@desktop';
import { withViewRaised } from '../../_view/Button_view_raised@desktop';

// _width
import { withWidthAuto } from '../../_width/Button_width_auto';
import { withWidthMax } from '../../_width/Button_width_max';

export * from '../../Button@desktop';

export const Button = compose(
    composeU(
        composeU(
            withPinBrickBrick,
            withPinBrickCircle,
            withPinBrickClear,
            withPinBrickRound,
            withPinCircleBrick,
            withPinCircleCircle,
            withPinCircleClear,
            withPinClearBrick,
        ),
        withPinClearCircle,
        withPinClearClear,
        withPinClearRound,
        withPinRoundBrick,
        withPinRoundClear,
    ),
    composeU(withSizeL, withSizeM, withSizeS),
    composeU(
        withThemeAction,
        withThemeClear,
        withThemeLink,
        withThemeNormal,
        withThemePseudo,
        withThemeRaised,
        withThemeWebSearch,
    ),
    composeU(withTypeLink, withTypeSubmit),
    composeU(withWidthAuto, withWidthMax),
    composeU(withViewDefault, withViewAction, withViewClear, withViewLink, withViewPseudo, withViewRaised),
    withBaseline,
)(ButtonDesktop);

export type IButtonProps = ExtractProps<typeof Button>;
