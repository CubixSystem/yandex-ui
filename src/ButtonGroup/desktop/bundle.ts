import { compose, composeU, ExtractProps } from '@bem-react/core';

import {
    withGapS,
    withGapM,
    withGapL,
    withGapXL,
    withPinCircle,
    withPinRound,
    ButtonGroup as ButtonGroupDesktop,
} from '..';

export * from '..';

export const ButtonGroup = compose(
    composeU(withGapS, withGapM, withGapL, withGapXL),
    composeU(withPinCircle, withPinRound),
)(ButtonGroupDesktop);

export type ButtonGroupProps = ExtractProps<typeof ButtonGroup>;
