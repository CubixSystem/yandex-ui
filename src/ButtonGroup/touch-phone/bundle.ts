import { compose, composeU, ExtractProps } from '@bem-react/core';

import {
    withGapS,
    withGapM,
    withGapL,
    withGapXL,
    withPinCircle,
    withPinRound,
    ButtonGroup as ButtonGroupTouchPhone,
} from '..';

export * from '..';

export const ButtonGroup = compose(
    composeU(withGapS, withGapM, withGapL, withGapXL),
    composeU(withPinCircle, withPinRound),
)(ButtonGroupTouchPhone);

export type ButtonGroupProps = ExtractProps<typeof ButtonGroup>;
