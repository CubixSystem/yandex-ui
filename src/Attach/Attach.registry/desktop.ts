import { Registry } from '@bem-react/di';

import { cnAttach } from '../Attach.const';
import { Holder } from '../Holder/Attach-Holder';
import { Control } from '../Control/Attach-Control';

export const attachRegistry = new Registry({ id: cnAttach() });

attachRegistry
    .set('Holder', Holder)
    .set('Control', Control)
    .set('Button', () => {
        // TODO: Использовать error api из новой версии react.
        throw new Error(`Для работы "Attach" требуется установить компонент "Button" в реестр "${cnAttach()}".`);
    });
