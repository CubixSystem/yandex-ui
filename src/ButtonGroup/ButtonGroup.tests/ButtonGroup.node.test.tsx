/**
 * @jest-environment node
 */
import React from 'react';
import { render } from 'enzyme';

import { ButtonGroup } from '../desktop/bundle';
import { Button } from '../../Button/desktop/bundle';

describe('ButtonGroup@dekstop', () => {
    test('должен вернуть полный шаблон компонента (snapshot)', () => {
        expect(
            render(
                <ButtonGroup>
                    <Button />
                </ButtonGroup>,
            ),
        ).toMatchSnapshot();
    });
});
