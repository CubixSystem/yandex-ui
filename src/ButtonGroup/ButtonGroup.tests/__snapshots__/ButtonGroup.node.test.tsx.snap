// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup@dekstop должен вернуть полный шаблон компонента (snapshot) 1`] = `
<div
  class="ButtonGroup"
  role="group"
>
  <button
    autocomplete="off"
    class="Button2"
    type="button"
  />
</div>
`;
